### PR TITLE
Implement Go Server for CLI Command Handling ( resolves #727 )

### PR DESCRIPTION
**Pull request by rishavvajpayee**
## Changes Made:

- Created a functional Go server with initial structure

### CODE REVIEW

**Comment by rishavvajpayee** [comment]
@lucifercr07  hey prashant please review this PR. Let me know if any config or structure changes you'd want.

**Comment by lucifercr07** [comment]
@rishavvajpayee let's just build the struct as part of this, I see the issue https://github.com/DiceDB/dice/issues/727 has been assigned to someone else, it'd be unfair to them.
Can you please trim down the scope of the PR just creating initial struct, so that everyone gets a fair chance. You can collaborate with assignee of that issue for now.

**Comment by rishavvajpayee** [comment]
sure @lucifercr07 . Will trim the PR.

**Comment by rishavvajpayee** [comment]
@lucifercr07 trimmed the PR's scope | kept the code for creating initial struct

File: internal/api/api.go
Comment (lucifercr07) [inline internal/api/api.go:1]: Let's take inspiration from our existing httpServer implementation.
https://github.com/DiceDB/dice/blob/master/internal/server/httpServer.go
And try to keep this in same parity, also let's use `net/http` for now we can optimize with other libraries if we hit scale issues.

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package config
+
+import (
+	"os"
+
+	"github.com/joho/godotenv"
```
Comment (lucifercr07) [inline config/config.go:6]: Let's just use `os.Getenv` for now. 
We can set default values for addr.

If required we'll add support for config. Let's start with barebone.

**Review by lucifercr07 — CHANGES_REQUESTED** [review]
Thanks a lot for your effort, please check the comments once.

File: internal/api/api.go
Comment (rishavvajpayee) [reply internal/api/api.go:1]: done replaced  echo with net/http package, and took inspiration from the `Run` functionality from https://github.com/DiceDB/dice/blob/master/internal/server/httpServer.go. and rewote the server Run functionality with graceful shutdown using WaitGroups

File: cmd/main.go
[the remark this answers is not in the document]
Comment (rishavvajpayee) [reply cmd/main.go:1]: moved main.go to parent level

**Comment by rishavvajpayee** [comment]
applied the requested changes @lucifercr07. let me know if i missed anything. Thanks 👍

**Review by lucifercr07 — APPROVED** [review]
Thanks a lot for these changes @rishavvajpayee. Merging this.

**Pull request merged**